test/retry_test.go

```diff
@@ -107,9 +108,9 @@ func TestTaskRunRetry(t *testing.T) {
 	pods, err := c.KubeClient.Kube.CoreV1().Pods(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("Failed to list Pods: %v", err)
-	} else if len(pods.Items) != numRetries {
+	} else if len(pods.Items) != numRetries+1 {
```

Comment:
If this is going to be referenced twice, it should probably have a variable, and maybe even a comment. Something like (please feel free to rephrase):

```
// We expect N+1 Pods total, one for each failed and retried attempt, and one for the final attempt.
wantPods := numRetries + 1
```

Reply:
Done, thanks